### PR TITLE
Qstat -f fails for more than 48 job ids

### PR DESCRIPTION
**Pull request by Bhagat-Rajput**
#### Bug/feature Description
* qstat -f fails for more than 48 job ids due to lack of server connection thus qstat reports error
 "Too many open connections" 

#### Affected Platform(s)
* ALL

#### Cause / Analysis / Design
* Not closing the server connection in check_max_job_sequence_id() after using.

#### Solution Description
* Passed the already opened connection and qstat will close itself once work will be done.

#### Testing logs/output
* Testing logs : [qstat_after_fix.txt](https://github.com/PBSPro/pbspro/files/2990031/qstat_after_fix.txt)
[qstat_rstat_width_testing.txt](https://github.com/PBSPro/pbspro/files/2990033/qstat_rstat_width_testing.txt)
[ptl_qstat_qselect_after_fix.txt](https://github.com/PBSPro/pbspro/files/3018876/ptl_qstat_qselect_after_fix.txt)
[ptl_qstat_qselect_before_fix.txt](https://github.com/PBSPro/pbspro/files/3018877/ptl_qstat_qselect_before_fix.txt)
[valgrind_ptl_pbs_qstat_qselect.txt](https://github.com/PBSPro/pbspro/files/3018887/valgrind_ptl_pbs_qstat_qselect.txt)

#### Checklist:

- [x] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [x] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [x] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [ ] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
   - [ ] I have added documentation in the **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)**.
- [ ] I have added new **PTL test(s) to my commit**. (See **[using PTL for testing](https://pbspro.atlassian.net/wiki/display/DG/Using+PTL+for+Testing)**) *(or)*
   - [x] I have added  **manual test(s) to this pull request and explained why PTL is not appropriate** for this case.
- [x] All new and existing automated tests have passed. (See **[running automated PTL tests](https://pbspro.atlassian.net/wiki/display/DG/PTL+Quick+Start+Guide)**).
- [x] I have attached **test logs to this pull request** as evidence of testing/verification.

__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__

### CODE REVIEW

File: src/cmds/qstat.c [hunk 1]
```diff
@@ -2852,13 +2852,12 @@ qstat -B [-f] [-F format] [-D delim] [ server_name... ]\n";
...
-				    /* check the server attribute max_job_sequence_id value */
-					check_seqid_len = check_max_job_sequence_id("qstat");
-					if (check_seqid_len == 1) {
-						how_opt |= ALT_DISPLAY_INCR_WIDTH; /* increase column width*/
-					} else if(check_seqid_len == -1) {
-						fprintf(stderr, "qstat: Unable to fetch the width format\n");
-						exit(1);
+					/* check the server attribute max_job_sequence_id value */
+					if (p_server != NULL) {
+						check_seqid_len = check_max_job_sequence_id(p_server);
```
Comment (subhasisb) [inline src/cmds/qstat.c]: This is better, but will still call check_max_job_sequence_id() for every jobid, even though those jobs are from the same server. Thus, we should call this only once for one cnt2server() calls, way higher, where we do the pbs_statserver().

**Comment by RKORavi** [comment]
@Bhagat-Rajput could you please add time it takes to qstat 1500 job ids after the fix.

**Comment by Bhagat-Rajput** [comment]
Hello @RKORavi, i have just submitted 1500 sleep jobs on my VM and it took around ~9 secs to print the qstat -f output for all the 1500 jobs, time results are pasted below.
real	0m8.757s
user	0m1.265s
sys	0m5.051s
I think it could be better on actual performance machine.

**Comment by RKORavi** [comment]
> Hello @RKORavi, i have just submitted 1500 sleep jobs on my VM and it took around ~9 secs to print the qstat -f output for all the 1500 jobs, time results are pasted below.
> real	0m8.757s
> user	0m1.265s
> sys	0m5.051s
> I think it could be better on actual performance machine.

Thankyou for testing it.

**Comment by subhasisb** [comment]
@Bhagat-Rajput are u still working on this? If you are done, please remove the [WIP] flag from this. Also please run your tests under valgrind

**Comment by Bhagat-Rajput** [comment]
Hello @subhasisb , i've added a new ptl testcase and valgrind logs to the ticket.

**Pull request merged**